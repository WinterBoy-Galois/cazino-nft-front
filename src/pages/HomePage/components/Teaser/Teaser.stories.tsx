import React from 'react';
import { storiesOf } from '@storybook/react';
import Teaser from '.';

storiesOf('Pages/Home/Teaser', module).add('default', () => (
  <div className="container">
    <Teaser />
  </div>
));
