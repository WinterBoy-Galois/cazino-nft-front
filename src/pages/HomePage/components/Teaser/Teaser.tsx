import React from 'react';
import styles from './Teaser.module.scss';
import { useTranslation } from 'react-i18next';
import { TFunction } from 'i18next';

const renderTeaser = (trans: string, t: TFunction) => (
  <div
    dangerouslySetInnerHTML={{
      __html: t(trans, {
        highlight: `<span class=${styles.highlight}>`,
        '/highlight': '</span>',
        interpolation: { escapeValue: false },
      }),
    }}
  />
);

const Teaser: React.FC = () => {
  const { t } = useTranslation(['home']);

  return (
    <div className={styles.container}>
      <div className={`row`}>
        <div className={`col-12 col-md-6 ${styles.teaser}`}>
          <div className={styles.dot} />
          {renderTeaser('teaser.one', t)}
        </div>
        <div className={`col-12 col-md-6 ${styles.teaser}`}>
          <div className={styles.dot} />
          {renderTeaser('teaser.two', t)}
        </div>
      </div>
      <div className={`row`}>
        <div className={`col-12 col-md-6 ${styles.teaser}`}>
          <div className={styles.dot} />
          {renderTeaser('teaser.three', t)}
        </div>
        <div className={`col-12 col-md-6 ${styles.teaser}`}>
          <div className={styles.dot} />
          {renderTeaser('teaser.four', t)}
        </div>
      </div>
    </div>
  );
};

export default Teaser;
