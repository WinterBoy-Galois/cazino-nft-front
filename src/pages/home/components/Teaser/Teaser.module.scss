@import '../../../../styles/shared.scss';

.container {
  margin-top: 0.5rem;
  font-size: 18px;

  @include media-breakpoint-up(lg) {
    margin-top: 2rem;
    font-size: 30px;
  }
}

.teaser {
  margin-top: 1.5rem;
  display: flex;
  align-items: center;

  @include media-breakpoint-up(md) {
    margin-top: 4rem;
    align-items: flex-start;
  }

  &:first-of-type {
    @include media-breakpoint-up(xl) {
      padding-right: 3rem;
    }
  }

  &:last-of-type {
    @include media-breakpoint-up(xl) {
      padding-left: 3rem;
    }
  }
}

.highlight {
  font-weight: 600;
  color: $orange;
}

.dot {
  width: 29px;
  height: 29px;
  border-radius: 50%;
  background-color: $orange;
  flex-shrink: 0;
  margin: 0 1rem;

  @include media-breakpoint-up(md) {
    display: none;
  }
}
