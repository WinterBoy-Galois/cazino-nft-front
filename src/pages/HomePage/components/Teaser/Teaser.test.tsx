import React from 'react';
import { render } from '@testing-library/react';
import Teaser from '.';

describe('Teaser', () => {
  it('should match snapshot', () => {
    // Arrange

    // Act
    const container = render(<Teaser />);

    // Assert
    expect(container).toMatchSnapshot();
  });
});
