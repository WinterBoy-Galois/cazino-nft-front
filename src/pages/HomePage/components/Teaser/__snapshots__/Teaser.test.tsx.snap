// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Teaser should match snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="container"
      >
        <div
          class="row"
        >
          <div
            class="col-12 col-md-6 teaser"
          >
            <div
              class="dot"
            />
            <div>
              The 
              <span
                class="highlight"
              >
                time
              </span>
               has come to 
              <span
                class="highlight"
              >
                increase
              </span>
               quality of online crypto gambling
            </div>
          </div>
          <div
            class="col-12 col-md-6 teaser"
          >
            <div
              class="dot"
            />
            <div>
              We at 
              <span
                class="highlight"
              >
                CaZZZino
              </span>
               re-design and re-invent online games
            </div>
          </div>
        </div>
        <div
          class="row"
        >
          <div
            class="col-12 col-md-6 teaser"
          >
            <div
              class="dot"
            />
            <div>
              Online gambling industry goes through changes with the rise of 
              <span
                class="highlight"
              >
                new
              </span>
               technologies
            </div>
          </div>
          <div
            class="col-12 col-md-6 teaser"
          >
            <div
              class="dot"
            />
            <div>
              Challenge your 
              <span
                class="highlight"
              >
                luck
              </span>
               and 
              <span
                class="highlight"
              >
                win big
              </span>
              , while feeling 
              <span
                class="highlight"
              >
                secure
              </span>
               with fair gambling!
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="container"
    >
      <div
        class="row"
      >
        <div
          class="col-12 col-md-6 teaser"
        >
          <div
            class="dot"
          />
          <div>
            The 
            <span
              class="highlight"
            >
              time
            </span>
             has come to 
            <span
              class="highlight"
            >
              increase
            </span>
             quality of online crypto gambling
          </div>
        </div>
        <div
          class="col-12 col-md-6 teaser"
        >
          <div
            class="dot"
          />
          <div>
            We at 
            <span
              class="highlight"
            >
              CaZZZino
            </span>
             re-design and re-invent online games
          </div>
        </div>
      </div>
      <div
        class="row"
      >
        <div
          class="col-12 col-md-6 teaser"
        >
          <div
            class="dot"
          />
          <div>
            Online gambling industry goes through changes with the rise of 
            <span
              class="highlight"
            >
              new
            </span>
             technologies
          </div>
        </div>
        <div
          class="col-12 col-md-6 teaser"
        >
          <div
            class="dot"
          />
          <div>
            Challenge your 
            <span
              class="highlight"
            >
              luck
            </span>
             and 
            <span
              class="highlight"
            >
              win big
            </span>
            , while feeling 
            <span
              class="highlight"
            >
              secure
            </span>
             with fair gambling!
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
